src/python/gudhi/representations/vector_methods.py

```diff
@@ -806,32 +807,49 @@ def fit(self, X, y=None, sample_weight=None):
         Returns:
             self
         """
-        if not hasattr(self.quantiser, 'fit'):
-            raise TypeError("quantiser %s has no `fit` attribute." % (self.quantiser))
-
-        # In fitting we remove infinite death time points so that every center is finite
-        X = [dgm[~np.isinf(dgm).any(axis=1), :] for dgm in X]
+        n_clusters = self.quantiser.n_clusters
 
+        if not len(X):
+            raise Exception("Cannot fit Atol on empty target.")
+        measures_concat = np.concatenate(X)
         if sample_weight is None:
             sample_weight = [self.get_weighting_method()(measure) for measure in X]
-
-        measures_concat = np.concatenate(X)
         weights_concat = np.concatenate(sample_weight)
 
-        self.quantiser.fit(X=measures_concat, sample_weight=weights_concat)
+        # In fitting we remove infinite birth/death time points so that every center is finite. We do not care about duplicates.
```

Comment:
I am not sure if we should ask the user to do so before calling Atol (with `DiagramSelector(use=True, point_type="finite")` for instance), or if all representations that requires no infinite birth/death time points shall do the same. To be discussed.

Reply:
No strong opinion here too... @mglisse or @MathieuCarriere ?

Reply:
> Perhaps there is a case to be made for filtering-but-printing/warning if a vectorizer requiring no infinite death points receives one?

Warning if the filtering removed any infinite points looks like a good idea (I think, not 100% sure). (and here the default of warnings.warn to print the warning only once looks good, since this is about explaining that users may want to filter themselves before atol, not something specific to one dataset)

It is true that it would be cleaner not to filter in Atol, but it should be cheap enough, so for convenience, it doesn't really bother me.